Modules/Filtering/BinaryMathematicalMorphology/include/itkObjectMorphologyImageFilter.hxx

```diff
@@ -100,17 +100,11 @@ ObjectMorphologyImageFilter<TInputImage, TOutputImage, TKernel>::BeforeThreadedG
   {
     this->GetOutput()->FillBuffer(0);
   }
-}
 
-template <typename TInputImage, typename TOutputImage, typename TKernel>
-void
-ObjectMorphologyImageFilter<TInputImage, TOutputImage, TKernel>::DynamicThreadedGenerateData(
-  const OutputImageRegionType & outputRegionForThread)
-{
-  ImageRegionConstIterator<TInputImage> iRegIter;
-  ImageRegionIterator<TOutputImage>     oRegIter;
-  iRegIter = ImageRegionConstIterator<InputImageType>(this->GetInput(), outputRegionForThread);
-  oRegIter = ImageRegionIterator<OutputImageType>(this->GetOutput(), outputRegionForThread);
+  RegionType requestedRegion = this->GetOutput()->GetRequestedRegion();
+
+  auto iRegIter = ImageRegionConstIterator<InputImageType>(this->GetInput(), requestedRegion);
+  auto oRegIter = ImageRegionIterator<OutputImageType>(this->GetOutput(), requestedRegion);
```

Comment:
This is correct AAA style ([Auto Almost Always](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/)), I'm honestly still more used to the old (C++98/C++03) equivalent, when it is of the form `Type var(arg1, arg2)`:
```cpp
    ImageRegionConstIterator<InputImageType> iRegIter(this->GetInput(), requestedRegion);
    ImageRegionIterator<OutputImageType> oRegIter(this->GetOutput(), requestedRegion);
```
No request, just wondering, is ITK going all the way [AAA](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto)?